src/test/javascript/spec/component/competencies/prerequisite-form.component.spec.ts

```diff
@@ -1,103 +1,190 @@
+            declarations: [],
+            providers: [MockProvider(PrerequisiteService), MockProvider(LectureUnitService), { provide: TranslateService, useClass: MockTranslateService }],
         })
             .compileComponents()
             .then(() => {
-                componentFixture = TestBed.createComponent(PrerequisiteFormComponent);
-                component = componentFixture.componentInstance;
+                prerequisiteFormComponentFixture = TestBed.createComponent(PrerequisiteFormComponent);
+                prerequisiteFormComponent = prerequisiteFormComponentFixture.componentInstance;
             });
+        translateService = TestBed.inject(TranslateService);
     });
 
     afterEach(() => {
         jest.restoreAllMocks();
     });
 
-    it('should update description', () => {
-        componentFixture.detectChanges();
-        component.updateDescriptionControl('new description');
-
-        expect(component['form'].controls.description.getRawValue()).toBe('new description');
+    it('should initialize', () => {
+        prerequisiteFormComponentFixture.detectChanges();
+        expect(prerequisiteFormComponent).toBeDefined();
     });
 
-    it('should emit on submit', () => {
-        component.prerequisite = prerequisite;
-        componentFixture.detectChanges();
+    it('should submit valid form', fakeAsync(() => {
+        // stubbing competency service for asynchronous validator
+        const prerequisiteService = TestBed.inject(PrerequisiteService);
+        const getAllTitlesSpy = jest.spyOn(prerequisiteService, 'getCourseCompetencyTitles').mockReturnValue(of(new HttpResponse({ body: ['test'], status: 200 })));
 
-        component['form'].controls.title.setValue('new title');
-        const submitSpy = jest.spyOn(component.onSubmit, 'emit');
+        const competencyOfResponse: Competency = { id: 1, title: 'test' };
 
-        component.submit();
+        const response: HttpResponse<Competency[]> = new HttpResponse({
+            body: [competencyOfResponse],
+            status: 200,
+        });
 
-        prerequisite.title = 'new title';
-        expect(submitSpy).toHaveBeenCalledWith(prerequisite);
+        jest.spyOn(prerequisiteService, 'getAllForCourse').mockReturnValue(of(response));
+
+        prerequisiteFormComponentFixture.detectChanges();
+
+        const exampleTitle = 'uniqueName';
+        prerequisiteFormComponent.titleControl!.setValue(exampleTitle);
+        const exampleDescription = 'lorem ipsum';
+        prerequisiteFormComponent.descriptionControl!.setValue(exampleDescription);
+        const exampleLectureUnit = new TextUnit();
+        exampleLectureUnit.id = 1;
+
+        const exampleLecture = new Lecture();
+        exampleLecture.id = 1;
+        exampleLecture.lectureUnits = [exampleLectureUnit];
+
+        prerequisiteFormComponent.selectLectureInDropdown(exampleLecture);
```

Comment:
Shouldn't we be calling this on a CommonCourseCompetencyFormComponent instance? This method currently doesn't seem to be doing anything on the prerequisite form